app/util/DateOfBirthValidator.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2025 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package util
+
+import model.DateOfBirth
+
+import java.time.LocalDate
+import scala.util.{Failure, Success}
+
+case class DateOfBirthValidator() {
+
+  def isValidDate(dateOfBirth: DateOfBirth): Boolean =
+    dateOfBirth.asLocalDate().isSuccess
+
+  def isNotFutureDate(dateOfBirth: DateOfBirth): Boolean =
+    dateOfBirth.asLocalDate() match {
+      case Success(localDate) => localDate.isBefore(LocalDate.now())
+      case Failure(_)         => false
+    }
+}
```

Comment:
total other nitpick, not for now because it's consistent, but I don't think we're getting much value from these validator case classes - maybe they should go in DateOfBirth companion object or something like that

Reply:
@oscarduignan makes sense - good candidate for a refactor

Reply:
@oscarduignan done